File: src/main/java/me/miki/shindo/management/profile/mainmenu/impl/CustomBackground.java
package me.miki.shindo.management.profile.mainmenu.impl;

import me.miki.shindo.utils.animation.simple.SimpleAnimation;

import java.io.File;

public class CustomBackground extends Background {

    private final SimpleAnimation trashAnimation = new SimpleAnimation();
    private final File image;

    public CustomBackground(int id, String name, File image) {
        super(id, name);
        this.image = image;
    }

    public File getImage() {
        return image;
    }

    public SimpleAnimation getTrashAnimation() {
        return trashAnimation;
    }
}

File: src/main/kotlin/me/miki/shindo/management/profile/mainmenu/impl/CustomBackground.kt
package me.miki.shindo.management.profile.mainmenu.impl

import me.miki.shindo.ui.animation.v2.value.SimpleAnimation
import java.io.File

class CustomBackground(
    id: Int,
    name: String,
    private val image: File,
) : Background(id, name) {
    private val trashAnimation = SimpleAnimation()

    fun getImage(): File = image

    fun getTrashAnimation(): SimpleAnimation = trashAnimation
}
